### PR TITLE
[Draft] Refactoring

### PR DESCRIPTION
**Pull request by michel2323**

### CODE REVIEW

File: src/ProxAL.jl [hunk 1]
```diff
@@ -47,19 +47,19 @@ function update_primal_penalty(x::PrimalSolution, opfdata::OPFData,
         β = [opfdata.generators[g].ramp_agc for g=1:ngen]
         @views for t=2:T
             x.Zt[:,t] .= ((algparams.τ*primal.Zt[:,t]) .- dual.ramping[:,t] .-
-                            (algparams.ρ_t[:,t].*(+x.Pg[:,1,t-1] .- x.Pg[:,1,t] .+ x.St[:,t] .- β))
-                        ) ./  max.(algparams.zero, algparams.τ .+ algparams.ρ_t[:,t] .+ (modelinfo.obj_scale*modelinfo.weight_quadratic_penalty_time))
+                            (algparams.ρ_t*(+x.Pg[:,1,t-1] .- x.Pg[:,1,t] .+ x.St[:,t] .- β))
```
Comment (frapac) [inline src/ProxAL.jl:50]: Why do we have a `(+` right after the parenthesis?

File: src/Evaluators/ProxALEvalutor.jl [hunk 2]
```diff
@@ -183,20 +196,16 @@ function optimize!(nlp::ProxALEvaluator)
...
         if runinfo.initial_solve
             if runinfo.iter == 1
-                set_penalty!(algparams,
-                         length(opfdata.generators),
-                         0.0,
-                         0.0,
-                         modelinfo)
-                algparams.updateρ_t = false
-                algparams.updateρ_c = false
+                algparams.ρ_t = 0.0
```
Comment (frapac) [inline src/Evaluators/ProxALEvalutor.jl:206]: I like that, it's clearer

File: src/params.jl [hunk 3]
```diff
@@ -31,23 +31,21 @@ Specifies ProxAL's algorithmic parameters.
 | `nlpiterlim::Int` |  maximum number of NLP subproblem iterations | 100
 | `tol::Float64` |     tolerance used for ProxAL termination | 1.0e-4
 | `zero::Float64` |    tolerance below which is regarded as zero | 1.0e-8
-| `ρ_t::Any` |         AL parameters for ramp constraints (can be different for different constraints) | 1.0
-| `ρ_c::Any` |         AL parameters for ctgs constraints (can be different for different constraints) | 1.0
-| `maxρ_t::Float64` |  Maximum value of `ρ_t` | 1.0
-| `maxρ_c::Float64` |  Maximum value of `ρ_c` | 1.0
+| `θ_t::Float64` |     see [Formulation](@ref) | 1.0
+| `θ_c::Float64` |     see [Formulation](@ref) | 1.0
+| `ρ_t::Float64` |     AL parameter for ramp constraints | 1.0
```
Comment (frapac) [inline src/params.jl:36]: `AL parameter` -> `penalty weight` ?

File: test/blockmodel.jl [hunk 4]
```diff
@@ -28,17 +28,15 @@ load_file = joinpath(DATA_DIR, "mp_demand", "$(case)_oneweek_168")
...
-    modelinfo.init_opf = false
 
     # Algorithm settings
     algparams = AlgParams()
     algparams.parallel = false #algparams.parallel = (nprocs() > 1)
     algparams.verbose = 0
     algparams.mode = :coldstart
+    algparams.ρ_t = maxρ
+    algparams.ρ_c = maxρ
+    algparams.τ = 3maxρ
```
Comment (frapac) [inline test/blockmodel.jl:39]: I would be more explicit here: `3 * maxρ`

**Review by frapac — APPROVED** [review]
It overall looks good to me: I like the new way the penalty weights are defined!
I only have a few suggestions to improve the code.

File: test/blockmodel.jl (hunk 4 above)
Comment (AnirudhSubramanyam) [reply test/blockmodel.jl:39]: Done

File: src/params.jl (hunk 3 above)
Comment (AnirudhSubramanyam) [reply src/params.jl:35]: Changed it to `AL penalty weight`

File: src/ProxAL.jl (hunk 1 above)
Comment (AnirudhSubramanyam) [reply src/ProxAL.jl]: Fixed now.

**Pull request closed without merging**